pymatgen/io/openmm/utils.py

```
@@ -360,111 +360,182 @@ def parameterize_system(
-            charged_openff_mols = assign_charges_to_mols(
-                smile_strings,
-                partial_charge_method,
-                partial_charge_scaling,
-                partial_charges,
-            )
-            openff_topology = openff.toolkit.topology.Topology.from_openmm(topology, charged_openff_mols)
-            box_vectors = list(np.array(box[3:6]) - np.array(box[0:3])) * angstrom
-            openff_topology.box_vectors = box_vectors
-            system = openff_forcefield.create_openmm_system(
-                openff_topology,
-                charge_from_molecules=charged_openff_mols,
-                allow_nonintegral_charges=True,
-            )
-            return system
+        charged_off_mols = assign_charges_to_mols(
+            smile_strings=smile_strings,
+            partial_charges=partial_charges,
+            partial_charge_scaling=partial_charge_scaling,
+            partial_charge_method=partial_charge_method,
+        )
+        if force_field.lower() in basic_small_ffs:
+            if force_field.lower() == "sage":
+                sage = SMIRNOFFTemplateGenerator(molecules=charged_off_mols, forcefield="openff-2.0.0")
+                forcefield_omm.registerTemplateGenerator(sage.generator)
+            elif force_field.lower() == "gaff":
+                gaff = GAFFTemplateGenerator(molecules=charged_off_mols, forcefield="gaff-2.11")
+                forcefield_omm.registerTemplateGenerator(gaff.generator)
+        else:
+            if "sage" in force_field.lower():
+                sage = SMIRNOFFTemplateGenerator(molecules=charged_off_mols, forcefield=force_field.lower())
+                forcefield_omm.registerTemplateGenerator(sage.generator)
+            elif "gaff" in force_field.lower():
+                gaff = GAFFTemplateGenerator(molecules=charged_off_mols, forcefield=force_field.lower())
+                forcefield_omm.registerTemplateGenerator(gaff.generator)
+
```

How about 
```python
smirnoff_ff_names = SMIRNOFFTemplateGenerator.INSTALLED_FORCEFIELDS
gaff_ff_names = GAFFTemplateGenerator.INSTALLED_FORCEFIELDS
if  ff_name == "sage" or ff_name in smirnoff_ff_names:
   ff_name = "openff-2.0.0" if ff_name == "sage" else ff_name
   ... do sage stuff
elif ff_name == "gaff" or ff_name in gaff_ff_name:
   ff_name = "gaff-2.11" if ff_name == "gaff" else ff_name
   ... do gaff stuff
else:
  ... # something about the forcefield not being implemented?
```

---

I'd also recommend wrapping this in a function, maybe `assign_small_molecule_ff` so you can use it down below.

---

Do you think this should actually throw an Error rather than a Warning about the forcefield not being implemented?

---

I'd vote for an Error, someone might miss a warning in high-throughput. If a FF is not available, I would think that is something that needs to be addressed. What would your preference be?

---

Agreed, put in an error for now